diff -U 3 -H -d -r -N -- serd-0.18.0/src/writer.c serd-0.18.0.orig/src/writer.c
--- serd-0.18.0/src/writer.c	2012-09-07 18:12:02.886064145 +0100
+++ serd-0.18.0.orig/src/writer.c	2012-09-07 18:11:16.000000000 +0100
@@ -103,7 +103,7 @@
 } TextContext;
 
 static void
-error2(SerdWriter* writer, SerdStatus st, const char* fmt, ...)
+error(SerdWriter* writer, SerdStatus st, const char* fmt, ...)
 {
 	va_list args;
 	va_start(args, fmt);
@@ -219,7 +219,7 @@
 			size = 4;
 			c    = in & 0x07;
 		} else {
-			error2(writer, SERD_ERR_BAD_ARG, "invalid UTF-8: %X\n", in);
+			error(writer, SERD_ERR_BAD_ARG, "invalid UTF-8: %X\n", in);
 			const uint8_t replacement_char[] = { 0xEF, 0xBF, 0xBD };
 			len += sink(replacement_char, sizeof(replacement_char), writer);
 			return len;
@@ -364,7 +364,7 @@
 		switch (writer->syntax) {
 		case SERD_NTRIPLES:
 			if (serd_env_expand(writer->env, node, &uri_prefix, &uri_suffix)) {
-				error2(writer, SERD_ERR_BAD_CURIE,
+				error(writer, SERD_ERR_BAD_CURIE,
 				      "undefined namespace prefix `%s'\n", node->buf);
 				return false;
 			}
@@ -603,7 +603,7 @@
 		return SERD_SUCCESS;
 	}
 	if (serd_stack_is_empty(&writer->anon_stack)) {
-		error2(writer, SERD_ERR_UNKNOWN,
+		error(writer, SERD_ERR_UNKNOWN,
 		      "unexpected end of anonymous node\n");
 		return SERD_ERR_UNKNOWN;
 	}
