diff -U 3 -H -d -r -N -- sord-0.8.0.old/src/zix/tree.c sord-0.8.0/src/zix/tree.c
--- sord-0.8.0.old/src/zix/tree.c	2012-04-17 18:01:39.000000000 +0100
+++ sord-0.8.0/src/zix/tree.c	2012-05-31 00:09:18.000000000 +0100
@@ -29,6 +29,7 @@
 	ZixTreeNode*  root;
 	ZixComparator cmp;
 	void*         cmp_data;
+        size_t        size;
 	bool          allow_duplicates;
 };
 
@@ -51,6 +52,7 @@
 	t->root             = NULL;
 	t->cmp              = cmp;
 	t->cmp_data         = cmp_data;
+        t->size             = 0;
 	t->allow_duplicates = allow_duplicates;
 	return t;
 }
@@ -74,6 +76,12 @@
 	free(t);
 }
 
+size_t
+zix_tree_size(ZixTree* t)
+{
+        return t->size;
+}
+
 static void
 rotate(ZixTreeNode* p, ZixTreeNode* q)
 {
@@ -343,6 +351,8 @@
 		}
 	}
 
+        ++t->size;
+
 	return ZIX_STATUS_SUCCESS;
 }
 
@@ -358,6 +368,8 @@
 	if ((n == t->root) && !n->left && !n->right) {
 		t->root = NULL;
 		free(n);
+                --t->size;
+                assert(t->size == 0);
 		return ZIX_STATUS_SUCCESS;
 	}
 
@@ -481,6 +493,8 @@
 
 	free(n);
 
+        --t->size;
+
 	return ZIX_STATUS_SUCCESS;
 }
 
diff -U 3 -H -d -r -N -- sord-0.8.0.old/src/zix/tree.h sord-0.8.0/src/zix/tree.h
--- sord-0.8.0.old/src/zix/tree.h	2012-04-17 18:01:39.000000000 +0100
+++ sord-0.8.0/src/zix/tree.h	2012-05-31 00:09:25.000000000 +0100
@@ -53,6 +53,12 @@
 zix_tree_free(ZixTree* t);
 
 /**
+   Return the number of elements in @a t.
+*/
+size_t
+zix_tree_size(ZixTree* t);
+
+/**
    Insert the element @a e into @a t and point @a ti at the new element.
 */
 ZixStatus
